Automattic/wp-calypso
# chore(packages): Migrate from `babel-eslint` to `@babel/eslint-parser`

#### Changes proposed in this Pull Request

Migrate from `babel-eslint` to `@babel/eslint-parser`.

[`babel-eslint`](https://www.npmjs.com/package/babel-eslint) is deprecated:

![image](https://user-images.githubusercontent.com/975703/109621606-7c157500-7b3b-11eb-8a68-e7fffbfdd805.png)

The new package `@babel/eslint-parser` should be a drop-in replacement. However, we still have `babel-eslint` in our repo, as it is a dependency of `@wordpress/eslint-plugin@7.3.0` ([existing issue](https://github.com/WordPress/gutenberg/issues/29179)). I also cleaned up the config related to eslint parsing and made our babel config explicit.

#### Testing instructions

* Checkout this branch and verify `yarn lint:js` still works (it will take a few minutes).
* Verify unit tests still pass
